### DISCUSSION TITLE
How to: Creating and presenting a glossary from Wikibase - best practice?

### DISCUSSION THREAD

**Discussion by mrchristian** [semanticClimate/ipcc-report-cataloguing#4] [Ideas]
For the IPCC Glossary we want to create and present the glossary using Wikibase. 

But how to complete a number of the steps in unclear - so this is a call out for assistance and pointers to learning resources.

Below I will outline our goals, the workflow, and show examples of glossaries that we think are interesting and well made.

##### Wikibase as 'Terminology service gateway'?

a la EPA Guidance https://www.epa.gov/web-policies-and-procedures/web-standard-glossaries

- Replicate this IPCC Glossary of 931 terms into a Wikibase instance - IPCC Glossary https://apps.ipcc.ch/glossary/ 
- TDM glossary download and transfer to Wikibase (we have parts of this workflow)
- Have the Wikibase either only editible by a user group, or have a moderation process, or completely locked

- Have easy to edit glossary interface with instructsions for editing (editing might be restricted or content comes in only from a pipeline)
- Put in place a open peer review and editorial quality assurance process

##### Rendering of Glossary

- Create a clean navigable web version of the glossary
- Create a multi-format reading and book format presentation view of the glossary using HTML, or Markdown and Vivliostyle (or Quarto or another platform) on GitHub/Lab. Output formats: Paged web, scrolling web, PDF, eBook, interoperable display source, source code.

##### Documentation 

- Document the whole process so that others can use it for their glossary creations and make documentation visible to tool support systems, e.g., in Wikibase support.

#### Using the glossary

- Use the glossary to show which IPCC reports use the terms:
  - The IPCC provide some infomration on this in their glossary
  - We have a 'Work in Progress' data model to be able to represent the the IPCC reports in a Wikibase: Currently only a spreadsheet - it will be committed to a Wikibase ASAP in a minimum form (by end of Oct 2023) - https://docs.google.com/spreadsheets/d/13N-PIjG0Nx30eyYSOL7CwdK-6J4kvIG8/edit#gid=1327168095
- Find the terms in Wikidata and somehow connect the IPCC Glossary terms to equivalents in Wikidata, and make it apparent in Wikidata that there is an IPCC Glossary and these are terms used in the glossary

### Workflows

This needs developing as the pathway is unclear, I will add what I know. SW

- TDM extraction from IPCC Glossary
- Inclusion in a Wikibase instance
- Render as a well designed, good looking, good UX, Glossary
- Present in a multi-format book form - ideas are rendering from HTML or Markdown via vivliostyle tooling as this involves a minimal processing - only a source file needs presenting and GitLab/Hub Pages and Vivliostyle will do the rest. 
- Workflow to Wikidata?
- ACLs, peer review, QA?

### Questions and help needed

- good practice for creating a glossary and glossary terms in Wikidata and Wikibase
- what is the best place to get Wikibase support and documentation online, see roadmaps, releases etc.
- are there example of glossaries, taxonomies, ontologies, terminology services made with Wikibase, or in Wikidata?
- benchmarks for glossary creation, guidance. W3C, EPA, etc?
- A web platform for displaying a glossary directly based on Wikibase content
- Options to export Wikibase content as HTML or Markdown for rendering by Vivliostyle or Quarto on GitLab/Hub Pagesor some other similar tooling

#### Guidance 

EPA - https://www.epa.gov/web-policies-and-procedures/web-standard-glossaries

#### Wikimedia

- https://www.mediawiki.org/wiki/Category:All_extensions
- https://www.mediawiki.org/wiki/Manual:Glossary/de 

#### Other

- W3C - https://www.w3.org/Glossary

**Comment by petermr** [comment]
Great

**Reply by mrchristian** [reply]
working on it ;-)
